### ISSUE TITLE
Reconsider interface

### ISSUE THREAD

**Issue by KnutAM** [KnutAM/FerriteAssembly.jl#11]
While fixing #9 , it seems like the current approach of mutating the old state variables to the new one might not be the best approach. 
If, inside `FerriteAssembly`, we could support handling of state variables that would probably be good since we dictate the structure anyways. Several reasons for doing this change: 

- The solution in #10 might be slow for large elements (but not tested, might also be good with large chunk size), 
- With old and new it doesn't matter if the assembly is done twice, no accumulation occurs 
- We avoid copying the states in each iteration for nonlinear problems, only one copy is required per time step.

The disadvantage is (1) that we get one extra input in the interface and (2) that we have to work with two parallel state vectors in the assembly loops. But these seem like minor issues.

**Comment by KnutAM** [comment]
One option could be 
```julia
element_routine!(
    Ke::AbstractMatrix, re::AbstractVector, new_state,
    ae::AbstractVector, material, cellvalues, buffer)
```
where `Δt` is moved into `buffer`, in addition to a variable `old_state`. These can be queried as `get_timestep(buffer)` and `get_old_state(buffer)`. 
Info from `dh_fh` is also moved into `buffer`, such that we can call `dof_range(buffer, ::Symbol)` instead of  `dof_range(dh_fh, ::Symbol)`
Motivation would be that the following items should go in the function signature
- Arguments being modified (`Ke, re , new_state`)
- Arguments being differentiated wrt. (`ae`)
- Arguments being dispatched (`material`)
- Arguments always being used (`cellvalues`)

The remaining items stay in `buffer` with access functions.

**Comment by KnutAM** [comment]
Closed by #15
